File: leetcode/rust/minimum-operations-to-make-the-array-increasing.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let mut result = 0;
        let mut nums = nums.clone();

        for i in 1..nums.len() {
            let diff = nums[i - 1] - nums[i] + 1;

            if diff > 0 {
                nums[i] += diff;
                result += diff;
            }
        }

        result
    }
}

File: leetcode/python/minimum-operations-to-make-the-array-increasing.py
class Solution:
    def minOperations(self, nums: list[int]) -> int:
        result = 0

        for i in range(1, len(nums)):
            diff = nums[i-1] - nums[i] + 1

            if diff > 0:
                nums[i] += diff
                result += diff

        return result